enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,274 @@
+the managed cluster even if the managed cluster resides in an insecure environment.
+
+The proposal [hosted-deploy-mode](./../33-hosted-deploy-mode/README.md) introduced a new deploy mode `Hosted` to deploy 
+klusterlet outside of the managed cluster, but keeps the addons agents remaining on the managed cluster, this proposal 
+makes the addon framework support running addons agents outside of the managed cluster, which is defined as `Hosted` 
+mode deploy for addons agents.
+
+### Goals
+
+- Make addon-framework deploy addons agents outside of the managed cluster.
+- Define how to deploy an addon in hosted mode
+
+### Non-Goals
+
+- Implementation details of addon
+- Support deploying all addons in hosted mode. Not all addons can run in hosted mode(ex: an addon agent 
+needs to analyze some managed cluster node related information, which can only be collected in the managed cluster and 
+cannot be exposed outside the managed cluster).
+
+## Proposal
+
+We propose to define an annotation of the `ManagedClusterAddon` for addon deployers to indicate that the addon is 
+deployed in hosted mode and where agents deployments will be deployed. And provide a label for addon developers to 
+specify which manifests need to be deployed outside the managed cluster.
+
+### User Stories
+
+#### Story 1
+
+As a managed cluster administrator, I hope as few as possible workloads running in my cluster while the addon 
+functionality is not affected.
+
+#### Story 2
+
+As a hub cluster administrator, on the premise that the addon functionality is not affected, I do not want the addon 
+agent to talk to the hub cluster directly. 
+
+### Terms
+
+- `Hosted mode`: deploy addons agents deployments outside the managed cluster.
+- `Hosting cluster`: The cluster where the addons agents deployments runn. A Hosting cluster is a managed cluster of 
+the hub, which is to host the addon agent to manage another managed cluster's addon.
+- `Addon deployer`: A person or a component responsible for deploying an addon.
+
+### Constraints
+
+There are several constraints:
+- The managed cluster must be exposed to the hosting cluster
+- The managed cluster(Klusterlet) must be imported to hub in `Hosted` mode
+- The hosting cluster must be a managed cluster of the hub, hosting cluster should be managed by the hub
+- The hosting cluster must be the same cluster where the managed cluster klusterlet(registration-agent & work-agent) 
+runs
+- After the addon is deployed, the deployment mode cannot be switched
+
+## Design Details
+
+### For addon developers
+
+We propose to add a label `addon.open-cluster-management.io/hosted-manifest-location` for the addon agent manifests to 
+mark which manifests should be deployed where in Hosted mode:
+- No matter what the label's value is, all manifests will be deployed on the managed cluster in Default mode.
+
+- When the label does not exist: the manifest will be deployed on the managed cluster in Hosted mode
```

Comment:
Could we also have a specific value like `managed` that does this? I think a developer could want to specify that they are deliberately placing something on the managed cluster. If the only way is to not specify a label, then it's not clear if the behavior is on purpose or not.

Reply:
Great point, I agree. @qiujian16 WDYT?